software/deploy/deploy.go

```diff
@@ -181,11 +181,18 @@ func buildPushDockerImage(houstonClient houston.ClientInterface, c *config.Conte
 		Output:          true,
 		Labels:          deployLabels,
 	}
+	useShaAsTag := config.CFG.ShaAsTag.GetBool()
 	err = imageHandler.Build("", "", buildConfig)
 	if err != nil {
 		return err
 	}
-
+	sha := ""
+	if useShaAsTag {
+		sha, err = imageHandler.GetBuiltImageSha()
+		if (sha == "") || (err != nil) {
+			return fmt.Errorf("failed to get image sha: %w", err)
+		}
+	}
```

Comment:
```suggestion
```
nit: keeping usage and value fetching close together to simplify the logic, hence moved it down where it is being used